### PR TITLE
CLOUDP-98421 (2nd followup) More fixes for metrics

### PR DESCRIPTION
**Pull request by pstefek-mongo**
They say that third time's the charm.

Jira: https://jira.mongodb.org/browse/CLOUDP-98421 also covers https://jira.mongodb.org/browse/CLOUDP-98878
EVG:   https://evergreen.mongodb.com/version/613974403627e04472e45c55?redirect_spruce_users=true

### CODE REVIEW

**Comment by TomerYakir** [comment]
hmm.. looks like there's a merge conflict..

**Comment by pstefek-mongo** [comment]
Newest EVG (in progress): https://spruce.mongodb.com/version/613a28260ae606164b189bab/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

**Review by TomerYakir — APPROVED** [review]

File: proxy_session.go [hunk 1]
```diff
@@ -43,7 +43,7 @@ type MetricsHook interface {
 }
 
 type MetricsHookFactory interface {
-	NewHook(metricName, labelName, labelValue string) (MetricsHook, error)
+	NewHook(metricName string, curryWithLabels map[string]string) (MetricsHook, error)
```
Comment (TomerYakir) [inline proxy_session.go:46]: did we consider having a more specific interface here? (e.g. `metricName, operationType, subtype string`)

Comment (pstefek-mongo) [reply proxy_session.go:46]: The hook is for any histogram so I don't want to force something more specific on a metric where operation type doesn't make sense. Also operation type is not known ahead of time

**Pull request merged**